clippy_lints/src/methods/suspicious_splitn.rs

```diff
@@ -0,0 +1,52 @@
+use clippy_utils::diagnostics::span_lint_and_note;
+use if_chain::if_chain;
+use rustc_ast::LitKind;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::source_map::Spanned;
+
+use super::SUSPICIOUS_SPLITN;
+
+pub(super) fn check(
+    cx: &LateContext<'_>,
+    method_name: &str,
+    expr: &Expr<'_>,
+    self_arg: &Expr<'_>,
+    count_arg: &Expr<'_>,
+) {
+    if_chain! {
+        // Ignore empty slice literal
+        if !matches!(self_arg.kind, ExprKind::Array([]));
+        // Ignore empty string literal
+        if !matches!(self_arg.kind, ExprKind::Lit(Spanned { node: LitKind::Str(s, _), .. }) if s.is_empty());
+        if let ExprKind::Lit(count_lit) = &count_arg.kind;
+        if let LitKind::Int(count, _) = count_lit.node;
```

Comment:
Would it make sense to use ``clippy_utils::consts`` here, so that also constant expressions evaluating to ``0`` or ``1`` get linted?

Reply:
I imagine it would be unlikely to occur, but I see no reason not to. It could be a false positive with something cfg dependent, but that could just as likely be a bug.

Reply:
Ok, if you also can't think of a reason against it, let's lint on all constants. This is also the usual behavior of Clippy, so it should be fine here too.

Reply:
Done now.